addons/sale_timesheet/models/account.py

```diff
@@ -79,6 +80,18 @@ def _compute_partner_id(self):
     def _compute_project_id(self):
         super(AccountAnalyticLine, self.filtered(lambda t: t._is_not_billed()))._compute_project_id()
 
+    @api.depends_context('uid')
+    @api.depends('order_id')
+    def _compute_has_so_access(self):
+        sale_leads_group = self.user_has_groups('sales_team.group_sale_salesman_all_leads')
+        if sale_leads_group:
+            self.has_so_access = True
+            return
+        sale_man_group = self.user_has_groups('sales_team.group_sale_salesman')
+        for timesheet in self:
+            timesheet.has_so_access = sale_man_group and timesheet.order_id.sudo().user_id == self.env.user
```

Comment:
Adding directly group(`sales_team.group_sale_salesman_all_leads`) will not display stat button for user with `sales_team.group_sale_salesman` group his own SOs/Invoices or group(`sales_team.group_sale_salesman`) will display SOs/Invoices even though user does not have it's access. Is it okay? as we will be changing it's functionality :thinking:

Reply:
I will ask to @aju-odoo to be sure but since we just use `sales_team.group_sale_salesman` group in sale_project (you can check you will see some examples 🙂 ), I guess if we do that here to be coherent with what we did in project. 🙂

Reply:
And she agrees with me, you can do the same way than we did in project, that is just checked if the user has `sales_team.group_sale_salesman` group. 

And so we can remove that field. 🙂

Reply:
Done! Thank you :slightly_smiling_face: